### ISSUE TITLE
[REST API] Argument 2 passed to Magento\Eav\Model\Attribute\Data\Text::validateLength() must be of the type string, null given

### ISSUE THREAD

**Issue by TomashKhamlai** [magento/magento2#23588]
The server's response code is 500 in cases either success or failure when Show Prefix is set to be required. Customer is created by setting prefix in Address section

### Preconditions (*)

1. No preconditions

### Steps to reproduce (*)

1. Navigate to Stores -> Settings -> Configuration -> Customers -> Customers Configuration
2. Set **Show Prefix** to be **Required**
3. Reindex and Flush Cache
4. Send request:
```shell
### Set your endpoint name, example: endpoint="http://gql-ce.io/rest"
endpoint="REPLACE";

admin_token=$(curl -X POST "$endpoint/V1/integration/admin/token" \
 -H "Content-Type: application/json" \
 -d '{"username":"admin","password":"123123q"}') && echo $admin_token && admin_token=$(echo $admin_token | tr -d '"');

 curl -X POST $endpoint/V1/customers \
-H "Content-Type: application/json" \
-H "Authorization: Bearer $admin_token" \
-d '{"customer":{"default_billing":"1","default_shipping":"1","email":"yhitecwa12@gmail.com","firstname":"Yokoshima","lastname":"Hiteca","gender": "3","store_id":1,"website_id":1,"addresses":[{"id":1,"region":{"region_code":"CA","region":"California","region_id":12},"region_id":12,"country_id":"US","street":["6161 West Centinela Avenue"],"company":"Magento","telephone":"555-55-555-55","postcode":"90230","city":"Culver City","firstname":"Yokoshima","lastname":"Hiteca","default_shipping":true,"default_billing":true}],"disable_auto_group_change":0,"extension_attributes":{"is_subscribed":false}},"password":"MetaFrame1"}' | json_pp
```
or
```shell
### Set your endpoint name, example: endpoint="http://gql-ce.io/rest"
endpoint="REPLACE";

admin_token=$(curl -X POST "$endpoint/V1/integration/admin/token" \
 -H "Content-Type: application/json" \
 -d '{"username":"admin","password":"123123q"}') && echo $admin_token && admin_token=$(echo $admin_token | tr -d '"');

 curl -X POST $endpoint/V1/customers \
-H "Content-Type: application/json" \
-H "Authorization: Bearer $admin_token" \
-d '{"customer":{"default_billing":"1","default_shipping":"1","email":"yhitecwa12@gmail.com","firstname":"Yokoshima","lastname":"Hiteca","gender": "3","prefix":"Mr.","store_id":1,"website_id":1,"addresses":[{"id":1,"region":{"region_code":"CA","region":"California","region_id":12},"region_id":12,"country_id":"US","street":["6161 West Centinela Avenue"],"company":"Magento","telephone":"555-55-555-55","postcode":"90230","city":"Culver City","firstname":"Yokoshima","lastname":"Hiteca","default_shipping":true,"default_billing":true}],"disable_auto_group_change":0,"extension_attributes":{"is_subscribed":false}},"password":"MetaFrame1"}' | json_pp
```

### Expected result (*)

1. Validation
2. Response code `200` in case of success (Customer was created after the second request, but got 500 in the console).

### Actual result (*)

1. No validation. Code 500 on every request

**Comment by engcom-Bravo** [comment]
@magento give me 2.3-develop instance

**Comment by magento-engcom-team** [comment]
Hi @engcom-Bravo. Thank you for your request. I'm working on Magento 2.3-develop instance for you

**Comment by magento-engcom-team** [comment]
Hi @engcom-Bravo, here is your [Magento instance](https://i-23588-2-3-develop.instances.magento-community.engineering).
Admin access: [https://i-23588-2-3-develop.instances.magento-community.engineering/admin](https://i-23588-2-3-develop.instances.magento-community.engineering/admin)
Login: `admin` Password: `123123q`
Instance will be terminated in up to 3 hours.

**Comment by engcom-Bravo** [comment]
By editing previous request with prefix foolows:
```shell
## Set your endpoint name, example: endpoint="http://gql-ce.io/rest"
endpoint="REPLACE";
curl -X POST $endpoint/all/V1/customers" -H "accept: application/json" -H "Content-Type: application/json" -H "Authorization: Bearer f0rgqlhfg4memmr6ngwzlcyn9scgk7cl" -d "{\"customer\":{\"default_billing\":\"1\",\"default_shipping\":\"1\",\"email\":\"yhitecwass132@gmail.com\",\"firstname\":\"Yokoshima\",\"lastname\":\"Hiteca\",\"gender\": \"3\",\"prefix\":\"Mr.\",\"store_id\":1,\"website_id\":1,\"addresses\":[{\"id\":1,\"region\":{\"region_code\":\"CA\",\"region\":\"California\",\"region_id\":12},\"region_id\":12,\"country_id\":\"US\",\"street\":[\"6161 West Centinela Avenue\"],\"company\":\"Magento\",\"telephone\":\"555-55-555-55\",\"postcode\":\"90230\",\"city\":\"Culver City\",\"firstname\":\"Yokoshima\",\"lastname\":\"Hiteca\",\"prefix\":\"Mr.\",\"default_shipping\":true,\"default_billing\":true}],\"disable_auto_group_change\":0,\"extension_attributes\":{\"is_subscribed\":false}},\"password\":\"MetaFrame1\"}"
```
Customer created, but code 500 on every request

**Comment by magento-engcom-team** [comment]
:white_check_mark: Confirmed by @engcom-Bravo 
 Thank you for verifying the issue. Based on the provided information internal tickets `MC-17971`, `MC-17972` were created

**Issue Available**: @engcom-Bravo, _You will be automatically unassigned. Contributors/Maintainers can claim this issue to continue. To reclaim and continue work, reassign the ticket to yourself._

**Comment by geet07** [comment]
@magento give me 2.3-develop

**Comment by geet07** [comment]
@magento give me 2.3-develop instance

**Comment by magento-engcom-team** [comment]
Hi @geet07. Thank you for your request. I'm working on Magento 2.3-develop instance for you

**Comment by magento-engcom-team** [comment]
Hi @geet07, here is your [Magento instance](https://i-23588-2-3-develop.instances.magento-community.engineering).
Admin access: [https://i-23588-2-3-develop.instances.magento-community.engineering/admin](https://i-23588-2-3-develop.instances.magento-community.engineering/admin)
Login: `admin` Password: `123123q`
Instance will be terminated in up to 3 hours.

**Comment by edenduong** [comment]
@magento give me 2.3-develop instance

**Comment by magento-engcom-team** [comment]
Hi @edenduong. Thank you for your request. I'm working on Magento 2.3-develop instance for you

**Comment by sergiolemus** [comment]
@TomashKhamlai, @chohanbhagchand, If you include the `"prefix"` for the `"customer":` and the `"addresses":` the return code is `200` every time. The request is shown below:

```Bash
curl -v -s -X POST $endpoint/V1/customers \
-H "Content-Type: application/json" \
-H "Authorization: Bearer $admin_token" \
-d '{"customer":{"default_billing":"1","default_shipping":"1","email":"yhitecwa12@gmail.com","firstname":"Yokoshima","lastname":"Hiteca","gender": "3","prefix":"Mr.","store_id":1,"website_id":1,"addresses":[{"id":1,"region":{"region_code":"CA","region":"California","region_id":12},"region_id":12,"country_id":"US","street":["6161 West Centinela Avenue"],"company":"Magento","telephone":"555-55-555-55","postcode":"90230","city":"Culver City","firstname":"Yokoshima","lastname":"Hiteca","prefix":"Mr.","default_shipping":true,"default_billing":true}],"disable_auto_group_change":0,"extension_attributes":{"is_subscribed":false}},"password":"MetaFrame1"}' | json_pp
```

**Comment by sdzhepa** [comment]
The related internal Jira ticket `MC-23813` was closed as `non-reproducible` in `2.4-develop`.
It means that Magento team either unable to reproduce this issue using provided _Steps to Reproduce_ from the **_Description section_** on clean Magento instance or the issue has been already fixed in the scope of other tasks.

But if you still run into this problem on the latest `2.4-develop` please create new Issue with updated information/steps/preconditions in the **_Description section_**.
